[🐛 | Bug]: `test_sync_with_concurrent_reencryption` failed

### Version

2.15.0

### What platforms you were running our software when the bug happened ?

Linux

### What happened ?

`test_sync_with_concurrent_reencryption` failed

### Relevant output

```
____________________ test_sync_with_concurrent_reencryption ____________________
[gw1] linux -- Python 3.9.16 /home/runner/.cache/pypoetry/virtualenvs/parsec-cloud-dZ-E5qY_-py3.9/bin/python

frozen_clock = <MockClock, time=300.0000000, rate=0.0 @ 0x7fa606e9c0a0>
running_backend = RunningBackend(asgi_app=<tests.common.backend.AsgiOfflineMiddleware object at 0x7fa6043a2100>, backend=<parsec.backend.app.BackendApp object at 0x7fa60437a5c0>, addr=BackendAddr { url: "parsec://127.0.0.1:35173?no_ssl=true" })
alice_core = LoggedCore(config=CoreConfig(config_dir=PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw1/test_sync_with_concurrent_...e9610>, _backend_conn=<parsec.core.backend_connection.authenticated.BackendAuthenticatedConn object at 0x7fa604418c40>)
bob_user_fs = <parsec.core.fs.userfs.userfs.UserFS object at 0x7fa6042f5520>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fa607165af0>

    @pytest.mark.trio
    async def test_sync_with_concurrent_reencryption(
        frozen_clock,
        running_backend,
        alice_core,
        bob_user_fs,
        monkeypatch,
    ):
        # Go fast
        alice_core.device.time_provider.mock_time(speed=1000.0)
        bob_user_fs.device.time_provider.mock_time(speed=1000.0)
        monkeypatch.setattr("parsec.utils.BALLPARK_ALWAYS_OK", True)
    
        # Create a shared workspace
        wid = await create_shared_workspace(EntryName("w"), bob_user_fs, alice_core)
        alice_workspace = alice_core.user_fs.get_workspace(wid)
        bob_workspace = bob_user_fs.get_workspace(wid)
    
        # Alice creates a files, let it sync
        await alice_workspace.write_bytes("/test.txt", b"v1")
        async with frozen_clock.real_clock_timeout():
            await alice_core.wait_idle_monitors()
        await bob_user_fs.sync()
    
        # Freeze Alice message processing so she won't process `sharing.reencrypted` messages
        allow_message_processing = trio.Event()
    
        async def _mockpoint_sleep():
            await allow_message_processing.wait()
    
        monkeypatch.setattr(
            "parsec.core.messages_monitor.freeze_messages_monitor_mockpoint", _mockpoint_sleep
        )
    
        # Now Bob reencrypt the workspace
        reencryption_job = await bob_user_fs.workspace_start_reencryption(wid)
        await bob_user_fs.process_last_messages()
        total, done = await reencryption_job.do_one_batch()
        assert total == done  # Sanity check to make sure the encryption is finished
    
        # Alice modify the workspace and try to do the sync...
        await alice_workspace.write_bytes("/test.txt", b"v2")
        # Sync monitor will try and fail to do the sync of the workspace
        await frozen_clock.sleep_with_autojump(300)
        assert not alice_core.are_monitors_idle()
    
        # Now let Alice process the `sharing.reencrypted` messages, this should
        # allow to do the sync
        allow_message_processing.set()
        async with frozen_clock.real_clock_timeout():
            await alice_core.wait_idle_monitors()
    
        # Just make sure the sync is done
        await bob_workspace.sync()
        for workspace in (bob_workspace, alice_workspace):
            info = await workspace.path_info("/test.txt")
            assert not info["need_sync"]
>           assert info["base_version"] == 3
E           assert 2 == 3

tests/core/test_sync_monitor.py:657: AssertionError
```

--- a/tests/core/test_sync_monitor.py
+++ b/tests/core/test_sync_monitor.py
@@ -615,11 +615,26 @@ async def test_sync_with_concurrent_reencryption(
     alice_workspace = alice_core.user_fs.get_workspace(wid)
     bob_workspace = bob_user_fs.get_workspace(wid)
 
-    # Alice creates a files, let it sync
+    # Create the file empty and sync it before writting into it. Without this step,
+    # Alice's sync monitor will have both `/` folder manifest and `/test.txt` file
+    # manifest to sync, with no guarantee which one will be done first. Hence two
+    # different outcomes for the `/test.txt`'s version (depending on if the file
+    # got a minimal sync because the parent folder was synchronized first)
+    await bob_workspace.touch("/test.txt")
+    await bob_workspace.sync()
+    await alice_workspace.sync()
+
+    # Alice creates version 2 of the file and let it sync
     await alice_workspace.write_bytes("/test.txt", b"v1")
     async with frozen_clock.real_clock_timeout():
         await alice_core.wait_idle_monitors()
-    await bob_user_fs.sync()
+    await bob_workspace.sync()
+
+    # Sanity check
+    for workspace in (bob_workspace, alice_workspace):
+        info = await alice_workspace.path_info("/test.txt")
+        assert not info["need_sync"]
+        assert info["base_version"] == 2
 
     # Freeze Alice message processing so she won't process `sharing.reencrypted` messages
     allow_message_processing = trio.Event()
@@ -649,8 +664,10 @@ async def test_sync_with_concurrent_reencryption(
     async with frozen_clock.real_clock_timeout():
         await alice_core.wait_idle_monitors()
 
-    # Just make sure the sync is done
+    # Bob now retrieves alice's changes
     await bob_workspace.sync()
+
+    # Just make sure the sync is done
     for workspace in (bob_workspace, alice_workspace):
         info = await workspace.path_info("/test.txt")
         assert not info["need_sync"]

